### ISSUE TITLE
rsyslog is stopped too early during shutdown

### ISSUE THREAD

**Issue by fmore-xyz** [systemd/systemd#8763]
I'm using Leap 42.3 with rsyslog as logging daemon (journal is disabled).

It appears that some messages from some services are missing from /var/log/messages during shutdown which is due to the fact that rsyslog service is treated by systemd like any other services. Therefore it's killed before other services that can still log output.

I'm opening a bug against systemd because I currently dont see how to indicate that a service should be stopped after all others.

**Comment by poettering** [comment]
Well, rsyslog would have to do pretty much what journald does and support running during early and late boot phases, i.e. in phases where most of the system isn't up. And then it could exclude itself from the usual shutdown logic just like journald does it and instead rely on the final killing spree to fix it.

However, none of this comes for free. If a service needs to run before the system is fully up and when the system is already mostly shut down it needs to be written in a certain style, i.e. without implied access to /var and suchlike. i.e. it's not just a simple as patching around in the unit file, it would actually have to be changed codewise.

Long story short: fix rsyslog to do what journald does and all should be good.

**Comment by poettering** [comment]
Oh, and this is a bug and RFE tracker. If you have questions how to do something or need help, please always contact the systemd mailing list instead.

**Comment by mbiebl** [comment]
@fmoreau you can setup rsyslog to pull messages from the journal instead of using the ForwardToSyslog=yes mode.
This way, rsyslog should get early boot and late shutdown messages.
